scripts/west_commands/runners/core.py

```diff
@@ -557,15 +535,6 @@ def add_parser(cls, parser):
         else:
             parser.add_argument('-i', '--dev-id', help=argparse.SUPPRESS)
 
-        if caps.flash_addr:
-            parser.add_argument('--dt-flash', default=False, choices=_YN_CHOICES,
```

Comment:
OK, I just got to this commit while reviewing and see the earlier patch changing `default=False` has no effect in the PR at all, so I'm now I'm just confused why it's needed in the PR.

Reply:
The situation currently is that dt-flash is broken in some cases. The idea was to first have commits fixing it (the first three). Then, as in the fixed situation every user of the option set it to True, delete it.

I did it like that because I thought it would be easier to understand why I wanted to delete the option, and I can easily either squash everything into one commit to just remove dt-flash, or just drop the last one to just fix the handling of the option but keeping it. Sorry if instead of helping it just made everything more confusing.